ksql-streams/src/main/java/io/confluent/ksql/execution/streams/StreamSourceBuilder.java

```diff
@@ -164,15 +174,34 @@ private static PhysicalSchema getPhysicalSchema(final AbstractStreamSource strea
         .transformValues(new AddKeyAndTimestampColumns<>(rowKeyGenerator));
   }
 
+  private static TimestampExtractor timestampExtractor(
+      final KsqlConfig ksqlConfig,
+      final LogicalSchema sourceSchema,
+      final Optional<TimestampColumn> timestampColumn) {
+    final TimestampExtractionPolicy timestampPolicy = TimestampExtractionPolicyFactory.create(
+        ksqlConfig,
+        sourceSchema,
+        timestampColumn
+    );
+    final int timestampIndex = timestampColumn.map(TimestampColumn::getColumn)
+        .map(c -> sourceSchema.valueColumnIndex(c).orElseThrow(IllegalStateException::new))
+        .orElse(-1);
+    return timestampPolicy.create(timestampIndex);
+  }
+
```

Comment:
As above, I think this code can be simplified more.  Do we even need our own `TimestampExtractionPolicy` any more? Why not just have 

```java
   final TimestampExtractor extractor = TimestampExtractorFactory.create(ksqlConfig, sourceSchema, timestampColumn);
```

i.e. get rid of `TimestampExtractionPolicy`, rename `TimestampExtractionPolicyFactory` -> `TimestampExtractorFactory` and move this logic into that class and have it return `TimestampExtractor`.
```

This PR makes the code cleaner, which is awesome. If you make this change it will be cleaner still.

Reply:
Yeah I was thinking the same thing. Left it for a follow-up though, to keep this PR focused.